### DISCUSSION TITLE
[RESOLVED] How can I use the key from MetaMask?

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [foundry-rs/foundry#2443] [General]
I'm trying to play Ethernaut with forge, how can I change the sender address and import my key?

**Comment by onbjerg** [comment]
Can you be a bit more specific? What command do you want to use the key on? It's all documented here: https://book.getfoundry.sh/

- [`forge create`](https://book.getfoundry.sh/reference/forge/forge-create)
- [`forge script`](https://book.getfoundry.sh/reference/forge/forge-script)
- [`cast send`](https://book.getfoundry.sh/reference/cast/cast-send)

**Reply by CaledoniaProject** [reply]
I want to run `forge test` with the wallet from MetaMask

**Reply by onbjerg** [reply]
You can set the account that calls the test contract itself using the `--sender` flag. For calls *from* the test contract to other contracts, you would use [`vm.prank`](https://book.getfoundry.sh/cheatcodes/prank)

**Reply by CaledoniaProject** [reply]
Yes, I saw that configuration from `forge config`, but how can I specify my private key?

**Reply by onbjerg** [reply]
You can't use a private key for `forge test` because you don't need one. What are you trying to achieve? Again, if you just need to set the addresses, you can use the `--sender` flag on `forge test` or `vm.prank`, depending on what sender you want to set.

**Reply by CaledoniaProject** [reply]
I'm trying to hack the Ethernaut levels: https://ethernaut.openzeppelin.com/level/0xe13a4a46C346154C41360AAe7f070943F67743c9
So I need to use the key from MetaMask, should I use `forge script` instead?

**Reply by onbjerg** [reply]
For `forge test` you don't need the key, just set `--sender` to the address of your MetaMask wallet. `forge test` will not sign any transactions and send anything, it's exclusively for testing. If you want to deploy something you should use `forge script`.

**Comment by CaledoniaProject** [comment]
I got it done with `--rpc-url` and `--private-key` and `vm.startBroadcast()`
